scylla-server-rust/src/main.rs

```diff
@@ -0,0 +1,41 @@
+use std::sync::Arc;
+
+use axum::{http::Method, routing::get, Router};
+use scylla_server_rust::{
+    controller::{
+        self, data_type_controller, driver_controller, location_controller, node_controller,
+        run_controller, system_controller,
+    },
+    prisma::PrismaClient,
+    Database,
+};
+use tower_http::cors::{Any, CorsLayer};
+
+#[tokio::main]
+async fn main() {
+    let client: Database = Arc::new(PrismaClient::_builder().build().await.unwrap());
+
+    let app = Router::new()
```

Comment:
we should have different routers for all of our types, most of them just have one route rn, but in the interest of scalability we should be building it more modulely

Reply:
Since the app is ultimately one namespace with one state, one CORS, etc. I really would hesitate to create a second router.  I do not know the implications of creating multiple CORS and multiple state `Database` etc, but I worry that it will lead to increase memory usage and bugs.  

When would a second router become beneficial for example in the express scylla?

This is the closest to a way to do that, but it doesnt mention state so Idk what will happen there: https://docs.rs/axum/latest/axum/routing/struct.Router.html#method.merge

I guess I could also make a bunch of nested services.